dotCMS/src/main/java/com/dotmarketing/servlets/ImportExportXMLServlet.java

```diff
@@ -191,20 +189,22 @@ private void doUpload(HttpServletRequest request, HttpServletResponse response)
 			 */
 			if (importFile != null && importFile.getName().toLowerCase().endsWith(".zip")) {
 
-				InputStream in = new BufferedInputStream(new FileInputStream(importFile));
-				ZipInputStream zin = new ZipInputStream(in);
+				InputStream in = new BufferedInputStream(Files.newInputStream(importFile.toPath()));
+				zin = new ZipInputStream(in);
 				ZipEntry e;
 
 				while ((e = zin.getNextEntry()) != null) {
 					unzip(zin, e.getName());
 				}
-				zin.close();
+
 				importFile.delete();
 			}
 
 		} catch (IOException e) {
 			// TODO Auto-generated catch block
 			Logger.error(this,e.getMessage(),e);
+		} finally {
+			zin.close();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "zin" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Add a "try/catch" block for "close". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1989)

Reply:
done